src/it/process-jar/verify.groovy

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+assert new File(basedir, 'target/classes').exists();
+assert new File(basedir, 'target/classes/org/jenkinsci/tools/hpi/its').exists();
+assert new File(basedir, 'target/classes/org/jenkinsci/tools/hpi/its/HelloWorldBuilder.class').exists();
+assert new File(basedir, 'target/classes/org/jenkinsci/tools/hpi/its/HelloWorldBuilder$DescriptorImpl.class').exists();
+assert new File(basedir, 'target/classes/org/jenkinsci/tools/hpi/its/HelloWorldBuilder.stapler').exists();
+assert new File(basedir, 'target/classes/org/jenkinsci/tools/hpi/its/Messages.class').exists();
+assert !new File(basedir, 'target/process-jar/WEB-INF/lib/jackson-annotations-2.6.0.jar').exists();
+assert !new File(basedir, 'target/process-jar/WEB-INF/lib/jackson-core-2.6.3.jar').exists();
+assert !new File(basedir, 'target/process-jar/WEB-INF/lib/jackson-databind-2.6.3').exists();
+assert new File(basedir, 'target/process-jar.hpi').exists();
+assert new File(basedir, 'target/process-jar.jar').exists();
+
+assert new File(basedir, 'target/generated-sources/localizer/org/jenkinsci/tools/hpi/its/Messages.java').exists();
+
+content = new File(basedir, 'target/generated-sources/localizer/org/jenkinsci/tools/hpi/its/Messages.java').text;
+assert content.contains(" holder.format(\"it.msg\");");
+
+// TODO add some test on hpi file content
```

Comment:
So with a fix to the Shade plugin, the IT would be to verify that the Jackson lib is shaded and that the plugin classes are using it that way, right?

In the meantime, your IT seems to be running jarsigner, but :bug: it is not verifying that `target/process-jar.hpi!/WEB-INF/lib/process-jar.jar!/META-INF/KEY.SF` exists.